src/surface_header.F90

```diff
@@ -183,6 +183,15 @@ end function iNormal
     procedure :: normal => z_cone_normal
   end type SurfaceZCone
 
+  type, extends(Surface) :: SurfaceQuadric
+    ! Ax^2 + By^2 + Cz^2 + Dxy + Eyz + Fxz + Gx + Hy + Jz + K = 0
+    real(8) :: A, B, C, D, E, F, G, H, J, K
+  contains
+    procedure :: evaluate => quadric_evaluate
```

Comment:
I haven't poked around the ray tracing code much in quite a while, but it does seem to me that the specificity of these routine signatures to the surface type likely necessitates the use of switch case blocks elsewhere in the code. Is it possible to use generic `evaluate`, `distance` and `normal` method signatures for each surface type so that the ray tracing code doesn't have to inspect the surface type to determine the correct routine to call? Perhaps this isn't possible in Fortran, it just seems like the most OO thing to do.